packages/js/client/src/PolywrapClient.ts

```diff
@@ -1,96 +1,73 @@
   GetFileOptions,
   GetImplementationsOptions,
   InterfaceImplementations,
   InvokeOptions,
   InvokerOptions,
-  PluginRegistration,
   QueryOptions,
   SubscribeOptions,
   Subscription,
   Uri,
-  UriRedirect,
+  IUriRedirect,
   createQueryDocument,
   getImplementations,
   parseQuery,
   TryResolveUriOptions,
   IUriResolver,
-  GetManifestOptions,
   IUriResolutionContext,
   UriPackageOrWrapper,
   UriResolutionContext,
   getEnvFromUriHistory,
-  PluginPackage,
   QueryResult,
   InvokeResult,
 } from "@polywrap/core-js";
-import {
-  buildCleanUriHistory,
-  IWrapperCache,
-} from "@polywrap/uri-resolvers-js";
+import { buildCleanUriHistory } from "@polywrap/uri-resolvers-js";
 import { msgpackEncode, msgpackDecode } from "@polywrap/msgpack-js";
 import {
   DeserializeManifestOptions,
   WrapManifest,
 } from "@polywrap/wrap-manifest-types-js";
 import { Tracer, TracerConfig, TracingLevel } from "@polywrap/tracing-js";
-import { ClientConfigBuilder } from "@polywrap/client-config-builder-js";
 import { Result, ResultErr, ResultOk } from "@polywrap/result";
-
-export interface PolywrapClientConfig<TUri extends Uri | string = string>
-  extends ClientConfig<TUri> {
-  readonly tracerConfig: Readonly<Partial<TracerConfig>>;
-  readonly wrapperCache?: Readonly<IWrapperCache>;
-}
+import { ClientConfigBuilder } from "@polywrap/client-config-builder-js";
 
 export class PolywrapClient implements Client {
-  private _config: PolywrapClientConfig<Uri> = ({
-    redirects: [],
-    plugins: [],
-    interfaces: [],
-    envs: [],
-    tracerConfig: {},
-  } as unknown) as PolywrapClientConfig<Uri>;
+  private _config: PolywrapCoreClientConfig<Uri>;
 
   constructor(
-    config?: Partial<PolywrapClientConfig<string | Uri>>,
+    config?: Partial<PolywrapClientConfig>,
+    options?: { noDefaults?: false }
+  );
+  constructor(
+    config: Partial<PolywrapCoreClientConfig>,
+    options: { noDefaults: true }
+  );
+  constructor(
+    config:
+      | Partial<PolywrapClientConfig>
+      | undefined
+      | Partial<PolywrapCoreClientConfig>,
     options?: { noDefaults?: boolean }
   ) {
     try {
       this.setTracingEnabled(config?.tracerConfig);
 
       Tracer.startSpan("PolywrapClient: constructor");
 
-      const builder = new ClientConfigBuilder();
-
-      if (!options?.noDefaults) {
-        builder.addDefaults(config?.wrapperCache);
-      }
-
-      if (config) {
-        builder.add(config);
-      }
-
-      const sanitizedConfig = builder.build();
-
-      this._config = {
-        ...sanitizedConfig,
-        tracerConfig: {
-          consoleEnabled: !!config?.tracerConfig?.consoleEnabled,
-          consoleDetailed: config?.tracerConfig?.consoleDetailed,
-          httpEnabled: !!config?.tracerConfig?.httpEnabled,
-          httpUrl: config?.tracerConfig?.httpUrl,
-          tracingLevel: config?.tracerConfig?.tracingLevel,
-        },
-      };
-
-      this._validateConfig();
+      this._config = !options?.noDefaults
```

Comment:
This does not properly handle the `config === undefined` scenario, what should be done when that is the case?

Reply:
It actually handles it, but typing are a bit out of date, I've updated them now